Services/Dashboard/Block/classes/class.ilDashboardBlockGUI.php

```diff
@@ -97,6 +97,14 @@ protected function getListItemGroups(): array
     {
         $data = $this->loadData();
         $groupedCards = [];
+        $obj_ids = [];
+        foreach ($data as $title => $group) {
```

Comment:
Why do you use a new `foreach` loop when you can use the existing foreach loop below starting in line `108`?

Reply:
`preloadListGUIData` has to be called before `getListItemForDataDTO`.
I could call it individually before each DTO call, but that would cause a heavier workload trough multiple SQL calls than the duplicated `foreach`